AirBench/Controllers/BenchController.cs

```diff
@@ -36,33 +40,80 @@ public ActionResult Create(CreateBench bench)
             return View("Create", bench);
         }
 
+        [Authorize]
         public ActionResult Review(int id)
         {
             CreateReview myReview = new CreateReview();
             myReview.BenchId = id;
             return View("Review",myReview);
         }
+        [Authorize]
         [HttpPost]
         public ActionResult Review (CreateReview review)
         {
-            BenchRepository repository = new BenchRepository(context);
-            Review myReview = new Review(review.Rating, review.Description, 1, review.BenchId);
-            repository.InsertReview(myReview);
-            return RedirectToAction("Details");
+            CustomPrincipal currentUser = (CustomPrincipal)User;
+            if (ModelState.IsValid)
+            {
+                BenchRepository repository = new BenchRepository(context);
+                Review myReview = new Review(review.Rating, review.Description, currentUser.User.Id, review.BenchId);
+                repository.InsertReview(myReview);
+                return RedirectToAction("Index");
+            }
+            return View(review);
         }
 
         public ActionResult Index ()
         {
             BenchRepository repository = new BenchRepository(context);
             List<Bench> myBenches = repository.GetBenchList();
-            return View("Index", myBenches);
+            List<BenchList> myBenchList = new List<BenchList>();
+            foreach(var bench in myBenches)
+            {
+                var reviews = repository.GetReviewList(bench.Id);
+                BenchList currentBench = new BenchList();
+                currentBench.CreatorUserId = bench.CreatorUserId;
+                currentBench.Description = bench.Description;
+                currentBench.Latitude = bench.Latitude;
+                currentBench.Longitude = bench.Longitude;
+                currentBench.NumberOfSeats = bench.NumberOfSeats;
+                currentBench.NumberOfReviews = reviews.Count;
+                currentBench.Id = bench.Id;
+                decimal sum = 0;
+                foreach (var review in reviews)
+                {
+                    sum += (decimal)review.Rating;
+                }
+                if (reviews.Count != 0)
+                {
+                    currentBench.Rating = sum / reviews.Count;
+                }
+                else
+                {
+                    currentBench.Rating = 0;
+                }
```

Comment:
How about putting all of this initialization code into a `BenchList` constructor? That'd help to keep your action method more focused.